### ISSUE TITLE
A crazy issue using svelte2dts through svelte2tsx

### ISSUE THREAD

**Issue by frederikhors** [sveltejs/language-tools#910]
**Describe the bug**

I have this component:

```svelte
<script lang="ts">
  export let one = false;
  export let disabled = false;
  export let two = false;
  export let three = "";
  export let four = false;
  export let fivefive = false;
</script>

<div {disabled}>Hello!</div>
```

if I use [`svelte2dts`](https://github.com/firefish5000/svelte2dts) to create it's declaration types it works but I have this error using it:

`Argument of type 'string' is not assignable to parameter of type 'never'. ts(2345)`

![image](https://user-images.githubusercontent.com/41120635/112138565-f16fe500-8bd1-11eb-83b8-8f4390034780.png)

If instead I rename this line:

```diff
- export let fivefive = false;
+ export let five = false;
```

**it works!** It's crazy! I know!

![image](https://user-images.githubusercontent.com/41120635/112138763-3136cc80-8bd2-11eb-9adf-4373b2ea0ef9.png)

**I think `svelte2dts` is using `svelte2tsx` for conversion, right?**

**To Reproduce**

Component REPO: https://github.com/frederikhors/svslte2tsx-issue-on-never

```
- clone it
- npm install
- npm link
- cd your svelte ts project
- npm link svslte2tsx-issue-on-never
- use it like below
- open VSCode
- TADA the problem!
```

Use it like this:

```svelte
<Test on:click={() => window.location.reload()}>Reload</Test>
```

**Expected behavior**

I do not understand with in the first case this is what I got in VSCode for `on`:

```
(method) Svelte2TsxComponent<{ one?: boolean; disabled?: boolean; two?: boolean; three?: string; four?: boolean; fivefive?: boolean; }, {}, {}>.$on<never>(event: never, handler: (e: never) => any): () => void
```

and in the second case I get:

```
(method) Svelte2TsxComponent<{ one?: boolean; disabled?: boolean; two?: boolean; three?: string; four?: boolean; five?: boolean; }, { [evt: string]: CustomEvent<any>; }, {}>.$on<"click">(event: "click", handler: (e: CustomEvent<any>) => any): () => void
```

**System (please complete the following information):**

- OS: Windows 10
- IDE: VSCode
- Plugin/Package: `svelte2dts` through `svelte2tsx`

**Comment by dummdidumm** [comment]
Yes, `svelte2dts` uses `svelte2tsx`. I have to check why this would make a difference. It should throw a type error in both cases because you don't forward the `click` event in `Test.svelte`, `on:click` will therefore never fire.

**Comment by dummdidumm** [comment]
I investigated this a little and I really don't know what's causing this. My current guess is that this is some obscure TypeScript bug which occurs when there is a certain amount of props with a certain length.

**Comment by frederikhors** [comment]
Oh my! Should we open some kind of issue on typescript repo?

**Comment by dummdidumm** [comment]
Not yet, I'll try to narrow this down more to get a minimum reproducible independent of `svelte2dts`

**Comment by SystemParadox** [comment]
Possibly related to https://github.com/sveltejs/language-tools/issues/1377?
